_posts/2023-06-23-run-and-debug-rust-verification-harnesses-natively-on-vscode.md

```diff
@@ -0,0 +1,179 @@
+
+```rust
+#[kani::proof]
+pub fn stretched_rectangle_can_hold_original() {
+    let original = Rectangle { width: kani::any(), height: kani::any() };
+    let factor = kani::any();
+    if let Some(larger) = original.stretch(factor) {
+        assert!(larger.can_hold(&original));
+    }
+}
+```
+
+#### How Kani is used currently
+
+The current way of interacting with Kani is through the command line.
+Users invoke `cargo kani` and specify the harness they want to verify.
+Kani produces text-based output that tells you whether your verification attempt has succeeded or failed.
+
+```
+$ cargo kani --harness stretched_rectangle_can_hold_original
+# --snip--
+Check 2: verification::stretched_rectangle_can_hold_original.assertion.1
+         - Status: FAILURE
+         - Description: "assertion failed: larger.can_hold(&original)"
+         - Location: src/main.rs:36:13 in function verification::stretched_rectangle_can_hold_original
+# --snip--
+VERIFICATION:- FAILED
+```
+
+## Introducing the Kani VS Code Extension
+
+The Kani VS Code extension offers a hassle-free and seamless integration into Visual Studio Code, making it more convenient to write and debug harnesses.
+As you write Kani harnesses, the extension detects them and conveniently displays them within your testing panel.
+The extension offers detailed diagnostics, feedback about verification failures, error messages, and stack traces.
+This empowers our users to find bugs and verify their code quicker.
+You can install the extension from the [webpage](https://marketplace.visualstudio.com/items?itemName=model-checking.kani-vscode-extension) on VS Code marketplace or by searching for `Kani` in the extensions tab of your VS Code window.
+
+## Using the VS Code extension
+
+### Verify Kani harnesses
+
+With the extension, running Kani on a harness to verify it, is as simple as clicking a button.
+The following sections walk you through the first few actions you’ll need.
+We’ll walk you through using the extension’s core features to debug and finally verify the rectangle example mentioned above.
+
+#### View Kani harnesses
+
+As soon as your Rust package is opened using the Kani extension in a VS Code instance, you should see the Kani harnesses loaded as regular unit tests in the Testing Panel on the [primary side bar](https://code.visualstudio.com/api/ux-guidelines/sidebars#primary-sidebar) of VS Code.
+This is how the testing page looks like when you click on the panel.
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/vs-code-images/view-kani-harnesses.png" alt="view-kani-harnesses" />
+
+#### Run Kani harnesses
+
+You can then run your harnesses using the tree view by clicking the play button beside the harness that was automatically picked up by the Kani Extension.
+Once you run the harness using the extension, you are shown a failure banner if verification failed (or a green check mark if it succeeded).
```

Comment:
Both cases are equally important (i.e., the parentheses aren't appropriate). I'd consider swapping the order since you're going to discuss the "verification failed" case.
```suggestion
Once you run the harness using the extension, you are shown a green check mark if verification succeeded, or a failure banner if it failed.
```